sound/soc/sdca/sdca_functions.c

```diff
@@ -1222,6 +1222,92 @@ static int find_sdca_entity_ge(struct device *dev,
 	return -EINVAL;
 }
 
+static int find_sdca_entity_hide(struct device *dev, struct fwnode_handle *function_node,
+	struct fwnode_handle *entity_node, struct sdca_entity *entity)
+{
+	struct sdca_entity_hide *hide = &entity->hide;
+	unsigned int delay, num_af, af_list[SDCA_MAX_FUNCTION_COUNT];
+	int nval, ret;
+	int num_hid_desc, num_report_decs;
+	struct hid_descriptor hid_desc;
+	unsigned char *report_desc = NULL;
+
+	ret = fwnode_property_read_u32(entity_node,
+			"mipi-sdca-RxUMP-ownership-transition-maxdelay", &delay);
+	if (ret)
+		hide->max_delay = 0;
+	else
+		hide->max_delay = delay;
+
+	nval = fwnode_property_count_u32(entity_node, "mipi-sdca-HIDTx-supported-report-ids");
+	if (nval > 0) {
+		hide->num_hidtx_ids = nval;
+		hide->hidtx_ids = devm_kcalloc(dev, hide->num_hidtx_ids,
+							sizeof(*hide->hidtx_ids), GFP_KERNEL);
+		if (!hide->hidtx_ids) {
+			fwnode_handle_put(entity_node);
+			return -ENOMEM;
+		}
+		ret = fwnode_property_read_u32_array(entity_node,
+						"mipi-sdca-HIDTx-supported-report-ids",
+						hide->hidtx_ids,
+						hide->num_hidtx_ids);
+		if (ret < 0)
+			return ret;
+	} else {
+		hide->num_hidtx_ids = 0;
+		hide->hidtx_ids = NULL;
+	}
+
+	nval = fwnode_property_count_u32(entity_node, "mipi-sdca-HIDRx-supported-report-ids");
+	if (nval > 0) {
+		hide->num_hidtx_ids = nval;
+		hide->hidtx_ids = devm_kcalloc(dev, hide->num_hidtx_ids,
+							sizeof(*hide->hidtx_ids), GFP_KERNEL);
+		if (!hide->hidtx_ids) {
+			fwnode_handle_put(entity_node);
+			return -ENOMEM;
+		}
+		ret = fwnode_property_read_u32_array(entity_node,
+						"mipi-sdca-HIDRx-supported-report-ids",
+						hide->hidtx_ids,
+						hide->num_hidtx_ids);
+		if (ret < 0)
+			return ret;
+	} else {
+		hide->num_hidrx_ids = 0;
+		hide->hidrx_ids = NULL;
+	}
+
+	num_af = fwnode_property_count_u32(entity_node, "mipi-sdca-hide-related-audio-function-list");
+	if (num_af <= 0) {
+		dev_err(dev, "%pfwP: HIDE - audio function numbers list missing: %d\n",
+			entity_node, num_af);
+		return -EINVAL;
+	} else if (num_af > SDCA_MAX_FUNCTION_COUNT) {
+		dev_err(dev, "%pfwP: maximum number of audio function exceeded\n", entity_node);
+		return -EINVAL;
+	}
+
+	fwnode_property_read_u32_array(entity_node, "mipi-sdca-hide-related-audio-function-list",
+		&af_list[0], num_af);
+
+
+	/* register HID device */
+	num_hid_desc = fwnode_property_count_u8(function_node, "mipi-sdca-hid-descriptor");
+	fwnode_property_read_u8_array(function_node, "mipi-sdca-hid-descriptor", (u8 *)&hid_desc, num_hid_desc);
+	hide->hid_desc = hid_desc;
+
+	num_report_decs = fwnode_property_count_u8(function_node, "mipi-sdca-report-descriptor");
```

Comment:
Can you re-use the same variable for all these counts in the function?

Reply:
will fix